### ISSUE TITLE
scheduleSequence silently continues when exceptions are thrown in the builder

### ISSUE THREAD

**Issue by krzysztofpniak** [dubzzz/fast-check#3717]
## 🐛 Bug Report

scheduleSequence silently continues when exceptions are thrown in its builder function

## To Reproduce

```
it('should allow assertions in the builder', async () => {
    await fc.assert(
      fc.asyncProperty(fc.scheduler(), async s => {
        s.scheduleSequence([
          {
            label: 'someLabel',
            builder: async () => {
              expect(1).toEqual(2);
            },
          },
        ]);

        await s.waitAll();
      })
    );
  });
```

## Expected behavior

exception should be processed by fast-check and printed with counterexample

## Your environment

| Packages / Softwares | Version(s) |
| -------------------- | ---------- |
| fast-check           |  3.8.0 |
| node                 |     18       |

**Comment by dubzzz** [comment]
Actually it's an expected behaviour in versions 3.x. If you want to make sure your builder never issued a failing promise and has been able to go up to its end you may edit your test as follow:

```js
it("should allow assertions in the builder", async () => {
  await fc.assert(
    fc.asyncProperty(fc.scheduler(), async (s) => {
      let lastSeen = undefined;
      const sequenceTask = s.scheduleSequence([
        {
          label: "someLabel",
          builder: async () => {
            expect(1).toEqual(2);
          },
        },
      ]);

      await s.waitAll();
      const { done, faulty } = await sequenceTask;
      expect(done).toBe(true);
      expect(faulty).toBe(false);
    })
  );
});
```

I'm going to document this behaviour closely in my new documentation. At least for now, a failure occurring inside a sequence does not explode the `waitAll` (as any failure during any other failing scheduled function).

**Comment by dubzzz** [comment]
Closing for now, expected behaviour

**Comment by dubzzz** [comment]
Actually after some investigations, I'd recommend users to prefer `waitFor` over `waitAll` whenever they have a task to wait for (which is the case for sequences). Otherwise they might somehow wait forever (see recent bug fixes around 3.8.3 and model based testing) or not wait enough.
